File: python/숨바꼭질(1697).py
from collections import deque

N, K = map(int, input().split())

visited = [False] * 100001

def bfs(start):
    queue = deque()
    queue.append([start, 0])

    while queue:
        cur, time = queue.popleft()

        if cur == K:
            return time

        for x in (cur-1, cur+1, cur*2):
            if 0<= x <= 100000 and not visited[x]:
                visited[x] = True
                queue.append([x, time+1])

print(bfs(N))

File: js/숨바꼭질(1697).js
const input = require('fs').readFileSync('/dev/stdin').toString().trim().split('\n');

const [N, K] = input.split(' ').map(Number);
const visited = Array.from({length: 100001}, () => false);

function bfs(start) {
    const queue = [];
    queue.push([start, 0])

    while (queue.length) {
        const [cur, time] = queue.shift();

        if (cur === K) {
            return time;
        }

        for (let i of [cur-1, cur+1, cur*2]) {
            if (i >= 0 && i <= 100000 && !visited[i]) {
                visited[i] = true;
                queue.push([i, time+1]);
            }
        }
    }
}